packages/browser/data-table/InternalUncontrolledDataTable/hooks/useUncontrolledDataTable.ts

```diff
@@ -75,15 +75,24 @@ export const useUncontrolledDataTable = <T>({
     setSearchParams(searchParams);
     syncData(searchParams);
   };
+  const [keyword, setKeyword] = useState(
+    searchParams.get(UncontrolledDataTableURLParams.KEYWORD)
+  );
 
-  const onKeywordChange = (keyword: string) => {
-    if (keyword.trim())
-      searchParams.set(UncontrolledDataTableURLParams.KEYWORD, keyword);
+  const handleKeywordDebounced = (tempKeyword: string) => {
+    if (tempKeyword.trim())
+      searchParams.set(UncontrolledDataTableURLParams.KEYWORD, tempKeyword);
     else searchParams.delete(UncontrolledDataTableURLParams.KEYWORD);
     setSearchParams(searchParams);
     syncData(searchParams);
   };
 
+  const debounceValue = useDebounce(handleKeywordDebounced);
+
+  const onKeywordChange = (tempKeyword: string) => {
```

Comment:
Rename `tempKeyword` to something more meaningful (like `inputKeyword` or `newKeyword` or something better)